src/main/java/net/snowflake/client/core/SFJsonResultSet.java

```diff
@@ -92,6 +92,24 @@ public Object getObject(int columnIndex) throws SFException {
     }
   }
 
+  /**
+   * Sometimes large BIGINTS overflow the java Long type. In these cases, return a BigDecimal type
+   * instead.
+   *
+   * @param columnIndex the column index
+   * @return an object of type long or BigDecimal depending on number size
+   * @throws SFException
+   */
+  private Object getBigInt(int columnIndex) throws SFException {
```

Comment:
To match with Arrow format as well as current behavior, getLong() should still be returned wherever possible. To avoid exceptions when the long object overflows, we can return a BigDecimal object instead.